### PR TITLE
Adding logging types to aws_eks_cluster

### LINKED ISSUE #240: Adding logging types into aws_eks_clusters
## Detailed Description
This would add two additional attributes to aws_eks_cluster, enabled_logging_types and disabled_logging_types. These would both return a list of the types of logging that have been enabled/disabled in AWS.

## Context
This could be used to check the logging status of an EKS cluster and identify which types have been enabled/disabled.

## Possible Implementation
I've developed this locally so I'll create a branch with the proposed changes.

### PR DESCRIPTION
**Pull request by voojoe**
Signed-off-by: Joe McCrea <joe.mccrea@sap.com>

### Description

This change adds both enabled_logging_types and disabled_logging_types as attributes to aws_eks_cluster. These will return a list of the logging types which have been enabled/disabled within the resource.

### Issues Resolved

#240 

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [N/A] New functionality includes integration tests/controls
- [N/A] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_eks_cluster.rb [hunk 1]
```diff
@@ -43,6 +43,8 @@ def initialize(opts = {})
       @creating              = resp[:status] == 'CREATING'
       @deleting              = resp[:status] == 'DELETING'
       @tags                  = resp[:tags]
+      @enabled_logging_types = resp[:logging][:cluster_logging].select(&:enabled).map { |type| type[:types] }
+      @disabled_logging_types= resp[:logging][:cluster_logging].reject(&:enabled).map { |type| type[:types] }
```
Comment (rmoles) [inline libraries/aws_eks_cluster.rb:47]: Thanks for the PR @voojoe. It seems these fields may not be stubbed out by the AWS API. Updating to this should get us past the build failures:
```
      @enabled_logging_types = resp.dig(:logging, :cluster_logging)&.select(&:enabled)&.map { |type| type[:types] }
      @disabled_logging_types= resp.dig(:logging, :cluster_logging)&.reject(&:enabled)&.map { |type| type[:types] }
```

**Comment by skpaterson** [comment]
Many thanks for this contribution @voojoe!  Could you also enhance the corresponding unit test with a check for the new properties?  
https://github.com/inspec/inspec-aws/blob/master/test/unit/resources/aws_eks_cluster_test.rb

**Comment by voojoe** [comment]
> Many thanks for this contribution @voojoe! Could you also enhance the corresponding unit test with a check for the new properties?
> https://github.com/inspec/inspec-aws/blob/master/test/unit/resources/aws_eks_cluster_test.rb

Hi Stuart, just working on these now and will commit once complete.

File: libraries/aws_eks_cluster.rb (hunk 1 above)
Comment (voojoe) [reply libraries/aws_eks_cluster.rb]: Hi Ross, thanks for the suggestion, I've introduced these changes now.

**Comment by voojoe** [comment]
Hi @skpaterson those tests proved a bit tricky but I've checked them in now, thanks for the pointer to those tests.

**Pull request merged**